libs/wingsdk/src/target-sim/container.inflight.ts

```diff
@@ -194,6 +241,114 @@ export class Container implements IContainerClient, ISimulatorResourceInstance {
     }
   }
 
+  private async docker(
+    command: string,
+    args: string[],
+    options: DockerSpawnOptions = {}
+  ): Promise<string> {
+    const child = this.dockerSpawn(command, args, options);
+    return child.join();
+  }
+
+  private dockerSpawn(
+    command: string,
+    args: string[],
+    options: DockerSpawnOptions = {}
+  ): DockerProcess {
+    let logStdout = options.stdout ?? false;
+    const logStderr = options.stderr ?? false;
+    const onError = options.onError;
+
+    // can be used to hide container logs (used in our end to end tests)
+    if (process.env.WING_HIDE_CONTAINER_LOGS) {
+      logStdout = false;
+    }
+
+    this.addTrace(
+      `Running: docker ${command} ${args.join(" ")}`,
+      TraceType.RESOURCE,
+      LogLevel.VERBOSE
+    );
+
+    const child = spawn("docker", [command, ...args], {
+      cwd: this.props.cwd,
+      stdio: "pipe",
+    });
+
+    let started = true;
+
+    child.once("exit", (code) => {
+      started = false;
+
+      if (onError && code !== 0 && code != null) {
+        onError(new Error(`non-zero exit code: ${code}`));
+      }
+    });
+
+    const output: Buffer[] = [];
+
+    child.stdout.on("data", (data) => {
+      output.push(data);
+
+      if (logStdout) {
+        this.addTrace(data.toString(), TraceType.LOG, LogLevel.INFO);
+      }
+    });
+
+    if (logStderr) {
+      child.stderr.on("data", (data) => {
+        this.addTrace(data.toString(), TraceType.LOG, LogLevel.INFO);
+      });
+    }
+
+    child.on("error", (err) => {
+      started = false;
+
+      if (logStderr) {
+        this.addTrace(err.stack ?? err.message, TraceType.LOG, LogLevel.ERROR);
+      }
+
+      if (onError) {
+        onError(err);
+      }
+    });
+
+    return {
+      get started() {
+        return started;
+      },
+      async kill() {
+        return new Promise((ok, ko) => {
+          if (!started) {
+            return ok();
+          }
+
+          child.kill("SIGKILL");
+          child.once("error", ko);
+          child.once("exit", ok);
+        });
+      },
+      async join() {
```

Comment:
nit: Maybe just call this `output` or `stdout`?

Reply:
I think `join()` is the "proper" name for this [1], no? Because it blocks and waits for the child to finish.

[1]: https://stackoverflow.com/a/25391156